Spigot-API-Patches/0082-Basic-PlayerProfile-interface.patch

```diff
@@ -0,0 +1,224 @@
+From 6d38632790caddf5f4dec1cfdd35d01b3fc3bd6c Mon Sep 17 00:00:00 2001
+From: Aikar <aikar@aikar.co>
+Date: Mon, 15 Jan 2018 21:46:46 -0500
+Subject: [PATCH] Basic PlayerProfile interface
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/profile/PlayerProfile.java b/src/main/java/com/destroystokyo/paper/profile/PlayerProfile.java
+new file mode 100644
+index 00000000..5c92bbcb
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/profile/PlayerProfile.java
+@@ -0,0 +1,67 @@
++package com.destroystokyo.paper.profile;
++
++import javax.annotation.Nonnull;
++import javax.annotation.Nullable;
++import java.util.Collection;
++import java.util.Set;
++import java.util.UUID;
++
++/**
++ * Represents a players profile for the game, such as UUID, Name, and textures.
++ */
++public interface PlayerProfile {
++
++    /**
++     * @return The players name, if set
++     */
++    @Nullable String getName();
```

Comment:
What about using `Optional<T>` instead of Nullable? Or is that not something that is usually done in the API?

Reply:
We do not use Optional in Bukkit/PAper

Reply:
Bukkit doesn't use Optionals, Nullable annotation has the same effect of "hey this might be null" in an IDE without the additional bloat of optional